scripts/generate-theme.ts

```diff
@@ -0,0 +1,56 @@
+import { readFileSync, writeFileSync } from 'fs';
+import { join, resolve } from 'path';
+import {
+  convert,
+  Prop,
+  registerFormat,
+  registerTransform,
+  registerValueTransform,
+} from 'theo';
+
+const npmLifecycleEvent = process.env.npm_lifecycle_event;
+
+if (!npmLifecycleEvent)
+  throw new Error('Unable to retrieve npm lifecycle event');
+
+const theme = npmLifecycleEvent.split(':').pop();
+
+const rootPath = resolve('.');
+const themesPath = join(rootPath, 'packages/core/src/theme/themes');
+
+const srcPath = join(themesPath, `${theme}.json`);
+const destPath = join(themesPath, `${theme}.scss`);
+const formatPath = join(themesPath, 'theme.scss.hbs');
+
+registerValueTransform(
+  'ods/color/var',
+  (prop: Prop) => prop.get('type') === 'color',
+  (prop: Prop) => `var(${cssVar(prop)}), ${opacity(prop)}`
+);
+
+registerTransform('raw', ['ods/color/var']);
+
+registerFormat('scss', readFileSync(formatPath, 'utf8'));
+
+convert({
+  transform: {
+    type: 'raw',
+    file: srcPath,
+  },
+  format: {
+    type: 'scss',
+  },
+})
+  .then((data) =>
+    writeFileSync(destPath, data.replace(/\$\$THEME/g, `${theme}`))
```

Comment:
`$$THEME` is an interesting choice, usually handlebars interpolations are like `{{THEME}}`.

I don't mind either, just curious 🤔

Reply:
So I can't use `{{THEME}}` as it gets interpolated during the Theo run.

I do need to replace this after Theo does it's job.

I don't mind having a different naming, `$$THEME`  was just some random choice...

Reply:
A-ha! I thought it was Theo doing the interpolation, and this was some sort of instruction. Makes more sense now.

I'm fine with `$$`. Although, can we not have Theo interpolate the theme? Would it require adding a new transform?

Reply:
I've replacing now before passing format source to Theo.